api/routes/slideshow.js

```diff
@@ -26,14 +27,47 @@ router
   })
   .delete('/:id', (req, res, next) => {
     const { id } = req.params
-    return Slideshow.findById(id)
+    return Slideshow.findByIdAndDelete(id)
       .then(slideshow => {
         if (!slideshow) return next('Slideshow not found')
-        return slideshow.remove().then(() => {
-          return res.send('success')
-        })
+        /**
+         * return slideshow.remove().then(() => {
+         * return res.send('success')
+         *})
+         */
+        deleteSlides(slideshow.slides)
```

Comment:
This needs a return here (as well as inside `deleteSlides`) and need to be able to send `res.send('success')`